src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c

```diff
@@ -21,47 +47,573 @@ const ActorInit Bg_Po_Event_InitVars = {
+    DECR(this->timer);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7328.s")
+    if (((this->timer == 0) || ((thisx->xzDistFromLink < 150.0f) && (thisx->yDistFromLink < 50.0f)) ||
+         (func_8002DD78(player) && (thisx->xzDistFromLink < 320.0f) &&
+          ((this->index != 2) ? (thisx->yDistFromLink < 100.0f) : (thisx->yDistFromLink < 0.0f)) &&
+          func_8002DFC8(thisx, 0x2000, globalCtx))) &&
+        ((thisx->parent != NULL) || (thisx->child != NULL))) {
+        /*The third condition in the || is checking if
+            1) Link is holding a ranged weapon
+            2) Link is too close in the xz plane
+            3) Link is too close in the y direction. The painting
+               under the balcony allows him to be closer.
+            4) Link is within 45 degrees of facing the painting. */
+        this->timer = 0;
+        Audio_PlayActorSound2(thisx, NA_SE_EN_PO_LAUGH);
+        this->actionFunc = BgPoEvent_PaintingVanish;
+    } else if (this->collider.base.acFlags & 2) {
+        if (!BgPoEvent_NextPicture(this)) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PO_SISTERS, thisx->posRot.pos.x,
+                        thisx->posRot.pos.y - 40.0f, thisx->posRot.pos.z, 0, thisx->shape.rot.y, 0,
+                        thisx->params + ((this->type - 1) << 8));
+            func_800800F8(globalCtx, 0xC58, 0x50, thisx, 0);
+            func_80078884(0x4802);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7444.s")
+        } else {
+            Audio_PlayActorSound2(thisx, NA_SE_EN_PO_LAUGH2);
+            func_800800F8(globalCtx, 0xC58, 0x23, thisx, 0);
+        }
+        if (thisx->parent != NULL) {
+            thisx->parent->child = NULL;
+            thisx->parent = NULL;
+        }
+        if (thisx->child != NULL) {
+            thisx->child->parent = NULL;
+            thisx->child = NULL;
+        }
+        this->timer = 20;
+        this->actionFunc = BgPoEvent_PaintingBurn;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7500.s")
+void BgPoEvent_PaintingBurn(BgPoEvent* this, GlobalContext* globalCtx) {
+    Vec3f sp54;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7530.s")
+    this->timer--;
+    sp54.x = (Math_Sins(this->dyna.actor.shape.rot.y) * 5.0f) + this->dyna.actor.posRot.pos.x;
+    sp54.y = Math_Rand_CenteredFloat(66.0f) + this->dyna.actor.posRot.pos.y;
+    sp54.z = Math_Rand_CenteredFloat(50.0f) + this->dyna.actor.posRot.pos.z;
+    if (this->timer >= 0) {
+        if (this->type == 2) {
+            func_8002A6B8(globalCtx, &sp54, &sZeroVec, &sZeroVec, 0x64, 0, 0xFF, 0xFF, 0x96, 0xAA, 0xFF, 0x00, 0x00, 1, 9,
+                          1);
+        } else {
+            func_8002A6B8(globalCtx, &sp54, &sZeroVec, &sZeroVec, 0x64, 0, 0xC8, 0xFF, 0xFF, 0xAA, 0x32, 0x64, 0xFF, 1, 9,
```

Comment:
i can take care of this on my end if you want, but here is the prototype for this effect

`
void EffectSsDeadDb_Spawn(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f* accel, s16 scale, s16 scaleStep, s16 primR, s16 primG, s16 primB, s16 primA, s16 envR, s16 envG, s16 envB, s16 unused, s32 arg14, s16 playSound)
`
sp54 is the position for the effect, and the 1 at the end is a boolean for if it should play a burning disintegrating sound. so colors/scale/scaleStep as decimal and that last thing as true would help